tools/rna_tools/rnaformer/infer_rnaformer.xml

```
@@ -0,0 +1,198 @@
+<tool id="infer_rnaformer" name="@EXECUTABLE@" version="@TOOL_VERSION@" profile="22.05">
+    <description>Predict the secondary structure of an RNA with RNAformer</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements">
+        <requirement type="package" version="1.83">biopython</requirement>
+    </expand>
+    <command detect_errors="exit_code"><![CDATA[
+    mkdir -p './model' &&
+    wget -O './model/RNAformer_32M_state_dict_intra_family_finetuned.pth' 'https://ml.informatik.uni-freiburg.de/research-artifacts/RNAformer/models/RNAformer_32M_state_dict_intra_family_finetuned.pth'
+    &&
+    wget -O './model/RNAformer_32M_config_intra_family_finetuned.yml' 'https://ml.informatik.uni-freiburg.de/research-artifacts/RNAformer/models/RNAformer_32M_config_intra_family_finetuned.yml'
+    &&
+    python '$script_file' > '$output'
+]]></command>
+<configfiles>
+        <configfile name="script_file"><![CDATA[import RNAformer
+import os
+import argparse
+import torch
+import urllib.request
+import logging
+from collections import defaultdict
+import torch.cuda
+import loralib as lora
+from RNAformer.model.RNAformer import RiboFormer
+from RNAformer.utils.configuration import Config
+
+from Bio import SeqIO
+
+import logging
+import sys
+
+config_file_path = 'model/RNAformer_32M_config_intra_family_finetuned.yml'
+model_file_path = 'model/RNAformer_32M_state_dict_intra_family_finetuned.pth'
+
+config = Config(config_file=config_file_path)
+config.RNAformer.cycling = 6
+model = RiboFormer(config.RNAformer)
+state_dict_file = model_file_path
+
+#if str($input_type.input_type) == 'True'
+fasta_path = '$input_type.fasta_input'
+sequences = [str(record.seq) for record in SeqIO.parse(fasta_path, 'fasta')]
+#else
+sequence_string = "$input_type.rna_input_string"
+sequences = [seq.strip() for seq in sequence_string.split(',')]
```

to you want to check here, if its really RNA sequences and fail early?